### PR TITLE
Add go-tools verification scripts 

### PR DESCRIPTION
**Pull request by ideahitme**
Scripts are mostly taken from the main kubernetes repository with slight changes. I added `verify-go-src.sh` script which runs each of the scripts inside repo-infra/go-tools directory. Normally user would just call `repo-infra/verify/verify-go-src.sh -v` from Travis (maybe indirectly) which in turn would run checks for `gofmt, gometalinter, govet` on all `.go` files excluding `/vendor`. Only added these checks to keep it project agnostic (e.g. exclude stuff like swagger/godep/bazel). 

/cc @eparis @pmorie

### CODE REVIEW

File: verify/verify-go-src.sh [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      ;;
+  esac
+done
+
+if ${SILENT} ; then
+  echo "Running in the silent mode, run with -v if you want to see script logs."
+fi
+
+ret=0
+run-checks "${GO_TOOLS_DIR}/*.sh" bash
```
Comment (eparis) [inline verify/verify-go-src.sh]: Doesn't this make it impossible to have included helper scripts? what was wrong with just running verify-*.sh?

**Comment by eparis** [comment]
how does this work if the repo is `vendor`d instead of git submoduled?

File: verify/verify-go-src.sh (hunk 1 above)
Comment (ideahitme) [inline verify/verify-go-src.sh]: In my mind, this leads to a mess when u start adding helper functions into the go-tools directory, which should be nothing more than slightly customised calls for go-tooling. I would prefer to have all verification scripts streamlined and serve their own purpose. Maybe there is something specific you have in my mind?

**Comment by ideahitme** [comment]
@eparis I added --rootdir flag so it can be used with vendoring.

**Comment by ideahitme** [comment]
@eparis https://github.com/kubernetes-incubator/external-dns/pull/29 - shows both way of integration

**Pull request merged**